### DISCUSSION TITLE
updating columns causes data to be rendered wrong

### DISCUSSION THREAD

**Discussion by dandelionn** [table-library/react-table-library#111] [Q&A] [answered]
Hello! If I create a table with 3 columns and then I modify the columns to 6 and after that I switch back to 3 columns the table still renders 6 columns. How can I fix this? 
I have explained the issue here: [updating columns causes data to be rendered wrong in react table library](https://stackoverflow.com/questions/76054148/updating-columns-causes-data-to-be-rendered-wrong-in-react-table-library)

**Comment by dandelionn** [comment, marked as answer]
I have found out that I need to modify this css variable:  "--data-table-library_grid-template-columns" depending on how many columns I want to display,
`
const columnsCount = 6;
getTableTheme = (columns) => {	
   return `
      --data-table-library_grid-template-columns: repeat(${columns}, minmax(0px, 1fr))
   `;
}

const theme = useTheme([
   getTheme(),
   {
      Table: getTableTheme(columnsCount),
      ...customTheme
   },
]);

return (
   <CompactTable
      theme={theme}
      ...
   />
)
`

**Comment by SillyColonel** [comment]
I have the same problem but this solution doesn't work for me. I might be doing something wrong. I've created a sandbox

https://codesandbox.io/p/sandbox/romantic-gianmarco-nq3nnw

The css-var in class applied to the table is updated when I remove columns after I've added them, but the css-var on the style applied on the table is lagging behind.
